internal/vpn/os_linux.go

```diff
@@ -42,3 +43,35 @@ func AddRoute(ip, gateway string) error {
 func DeleteRoute(ip, gateway string) error {
 	return run("ip", "r", "del", ip, "via", gateway)
 }
+
+var (
+	errEAGAIN error = syscall.EAGAIN
+	errEINVAL error = syscall.EINVAL
+	errENOENT error = syscall.ENOENT
+)
+
+// errnoErr returns common boxed Errno values, to prevent
+// allocations at runtime.
+func errnoErr(e syscall.Errno) error {
+	switch e {
+	case 0:
+		return nil
+	case syscall.EAGAIN:
+		return errEAGAIN
+	case syscall.EINVAL:
+		return errEINVAL
+	case syscall.ENOENT:
+		return errENOENT
+	}
+	return e
+}
+
+// Setuid sets uid of current OS user.
+func Setuid(uid int) (err error) {
+	// due to iss. 1435 `syscall.Setuid` is disabled for Linux system, so we'
```

Comment:
Could you please add the link (https://github.com/golang/go/issues/1435) to comment? That could be a bit clearer. Also, `so we'` looks unfinished.

Reply:
Not used anymore